### ISSUE TITLE
Timber Resize stretching images

### ISSUE THREAD

**Issue by njbarrett** [timber/timber#1435]
I'm having issues with Timber trying to resize my 100px image to 400px if I use the `|resize(400,0,'center')` filter.

I'm not sure if this is intended, because from my previous experience with Timber it did not try to "upscale" smaller images to fit the resize width and height.

Is there a way to tell Timber to not upscale smaller images?

**Comment by certainlyakey** [comment]
I second that... There should be at least option to set if you want to prevent images being upscaled.

**Comment by jarednova** [comment]
@njbarrett @certainlyakey I agree as well — do either of you have ideas on what that would look like in Twig? I'm hesitant to keep adding arguments to `resize` — is there another word that would make sense? `resize_but_dont_upscale` seems a bit much

**Comment by certainlyakey** [comment]
Hmm, I see no option other than adding one more argument. Introducing a new separate filter would probably complicate things, given that it would still need all the same arguments as `resize` currently has.

**Comment by njbarrett** [comment]
It depends whether the default functionality should be to upscale or not.. IMO it shouldn't but I guess a lot of sites could have been built with that in mind.
In a perfect world:
`|resize` - resizes but doesnt upscale
`|upscale` - resizes but does upscale

My 2 cents.

To get around this currently, I am doing an if statement to check the image size is not less than what im trying to resize to, e.g

```
                    {% if module.image.width > 800 %}
                        <img src="{{ module.image.src|resize(800, 0, 'center') }}" alt="{{ module.image.alt }}" title="{{ module.image.title }}" />
                    {% else %}
                        <img src="{{ module.image.src }}" alt="{{ module.image.alt }}" title="{{ module.image.title }}" />
                    {% endif %}
```

**Comment by FlyingDR** [comment]
At my sight there should be at least a way to control resizing process. For example [\Timber\Image\Operation\Resize::get_target_sizes()](https://github.com/timber/timber/blob/master/lib/Image/Operation/Resize.php#L85) may have Wordpress filters to decide if image should be upscaled. Something like:
```php
// ... calculation of target image size into get_target_sizes()
if (($w > $src_w || $h > $src_h) && (!apply_filters('timber/image/should_upscale', $image, $w, $h))) {
  // Return image size structure with default information
}
```
It will also be useful to have a filter to let developer to control  final image size / position, like:
```php
// ... target image size / position are calculated and stored in $result
return apply_filters('timber/image/resize', $image, $result);
```
These filters will give developers much better control over images resizing process

**Comment by palmiak** [comment]
I just ran into this - IMO I think that the 
```
|resize - resizes but doesnt upscale
|upscale - resizes but does upscale
```
is the best way to go.

Or maybe let's think about changing the last param to array - there are many things that could be changed with this (quality, upscale etc). And it would be great for backward compatibility - if user would provide a string if would work as it does now.

**Comment by matteopoile** [comment]
What about never upscale and use css if we need to fit a large container with small image?
